lib/spack/spack/directory_layout.py

```diff
@@ -88,34 +89,38 @@ def check_installed(self, spec):
         """
         raise NotImplementedError()
 
-    def extension_map(self, spec):
-        """Get a dict of currently installed extension packages for a spec.
+    def extension_map(self, spec, path_view=""):
```

Comment:
I am all for it: In fact, that was the first approach I considered, but since it prevented globally activated extensions to propagate into filesystem views I decided against it.

However, if we decide against globally activated extensions (which I would prefer!) this would be a non-issue and I'd get started right away.